### PR TITLE
Nav: add focus boundary

### LINKED ISSUE #32: Project template > Nav component: add focus boundary
# The problem
User focus can be lost within the mobile nav menu by tabbing past the last nav item.

## Why
People don't typically navigate mobile devices with keyboards; but we do need to develop UIs defensively, as there _are_ other methods of navigating, even on mobile devices.

## Potential solution
Hidden, focusable element at the end of the mobile menu, which, on focus, shifts focus back up to the close button.

### PR DESCRIPTION
**Pull request by andrewrubin**
# What this does
- adds a focus boundary (aka focus trap) to maintain a loop when tabbing through the expanded mobile navigation menu.
- refactors the breakpoint variable names to be cleaner, per the latest `useBreakpoints` hook updates.

Fixes #32

### CODE REVIEW

File: cli/templates/project/src/components/nav/nav.js
Comment (marlonmarcello) [inline cli/templates/project/src/components/nav/nav.js:1]: Completely unrelated but why do we need [this big](https://github.com/wethegit/corgi/pull/38/files#diff-cdf34e02b9c813798353cc889c3f182a6cbd96a720e5b8829f708879bcb73408R2) dependency just for the nav?

**Review by marlonmarcello — APPROVED** [review]

File: cli/templates/project/src/components/nav/nav.js
Comment (andrewrubin) [reply cli/templates/project/src/components/nav/nav.js:1]: @marlonmarcello only to animate the "mountedness", really. On some sites, I have opted for just always having the menu mounted, and opacity zero-ing it, or translating off the screen when not open. Happy to alter that if you like.

Comment (marlonmarcello) [reply cli/templates/project/src/components/nav/nav.js:1]: We already have NextJS and React which are beefy JS dependencies for such simple requirements. I feel like, any cut that we CAN make to our JS payload, we SHOULD make it. Unless it degrades UX.

Comment (marlonmarcello) [reply cli/templates/project/src/components/nav/nav.js:1]: But that can be on a separate PR @andrewrubin

Comment (andrewrubin) [reply cli/templates/project/src/components/nav/nav.js:1]: Yeah absolutely!

**Pull request merged**